Source/DafnyCore/Verifier/Translator.cs

```diff
@@ -3471,8 +3471,15 @@ public enum StmtType { NONE, ASSERT, ASSUME };
         if (node is IDeclarationOrUsage declarationOrUsage && node is not IHasUsages) {
           return new BoogieRangeToken(node.StartToken, node.EndToken, declarationOrUsage.NameToken);
         }
-        return node.RangeToken.ToToken();
+        return new BoogieRangeToken(node.StartToken, node.EndToken, node.Tok);
       } else {
+        // The commented line is what we want, but it changes what is translated.
+        // Seems to relate to refinement and possibly RefinementToken.IsInherited and or ForceCheckToken
+        // It might be better to remove calls to RefinementToken.IsInherited from this file, and instead
+        // add generic attributes like {:verify false} in the refinement phases, so that refinement does not complicate
+        // translation,
+        //
+        // return new BoogieRangeToken(node.StartToken, node.EndToken, node.Tok);
```

Comment:
That's precisely what I encountered. I think there might be two mechanism for reporting related ranges: One in nested token, another one in auxiliary information retrieved from Boogie. That might create a conflict that we solve one way or the other, but it's incompatible with the correct code above.
At least, given that we have tests in place, it should be enough for us to still have a decent behavior.